docs/storage/s3/parquet.md

```diff
@@ -0,0 +1,67 @@
+# Apache Parquet
+
+There is idea to use Apache Parquet for storing group relations between Runtime and the Application.
+Apache Parquet is a column-oriented data storage format, which could be used to hold the needed pieces of information. It would be stored on the root of the S3 bucket. 
+
+The whole structure on S3 would look like this:
+
+    ```
+    {tenant_id}/
+    ├── applications/
+    │   ├── {application_id}/
+    │   │  └── data.json
+    ├── runtimes/
+    │   ├── {application_id}/
+    │   │  └── data.json
+    └── db.parquet
+    ...
+    ```
+
+To minimize the file, the `db.parquet` file would store all details regarding labels for runtimes and applications. We would do querys for:
+- all labels
+- all applications with a specific label
+- all runtimes with a specific label
+- all labels with a specific key
+
+However, all read/write operations would need multiple S3 requests to handle them. For example:
+
+To read all runtimes (when we have 10 of them), we would need to:
+- fetch list of the `runtimes` directory (1 request)
+- for every runtime, we would need to download `data.json` file (10 requests)
+- download `db.parquet` file (1 request), query labels for all runtimes and merge the results above with data
+
+We could use similar approach to the one described [here](./README.md), by writing labels for runtimes and applications as files under `{tenant_id}/{applications or runtimes}/{application_id or runtime_id}/` directory.
+
+### Pros
+
+1. **Efficient data store**
+
+    Very good compression. For example, if CSV file is as big as 1 TB, the data stored in Parquet file has 130 GB.
+
+1. **Good performance**
+    
+    Fetching specific column values doesn't need reading the entire raw data.
+
+### Cons
```

Comment:
If for some reason our parquet file would be 100 GB large, and we wanted to apply any change to it, I think we would have to:
1. Download 100 GB file
2. Uncompress the file
3. Apply our changes
4. Compress the file
5. Upload 100 GB file to replace the old one

Even if we want to focus on optimizing read operations this seems to me like a pretty big problem, not only because we need to transfer large file two times over the internet, but also because of the implied space requirements. I'm also worried the compression/uncompression operations will be costly.

In case we have to uncompress the whole file, it would take 1 TB of management plane space during this process. If, for example, two or more users decide to modify a label at the same time this number will multiply.

According to answers in [this post](https://stackoverflow.com/questions/28837456/updating-values-in-apache-parquet-file/) it might be possible to uncompress only a part of parquet file but we would either need to partition the file (effectively use multiple parquet files instead of one) or use row groups.